src/main/java/org/team1540/robot2018/Robot.java

```diff
@@ -376,4 +204,199 @@ public void teleopPeriodic() {
+    SCALE_OWNED_RIGHT(() -> (MatchData.getOwnedSide(GameFeature.SCALE) == OwnedSide.RIGHT)),
+    SWITCH_OWNED_LEFT(() -> (MatchData.getOwnedSide(GameFeature.SWITCH_NEAR) == OwnedSide.LEFT)),
+    SWITCH_OWNED_RIGHT(() -> (MatchData.getOwnedSide(GameFeature.SWITCH_NEAR) == OwnedSide.RIGHT)),
+    NO_DATA(() -> (true));
+
+    @NotNull
+    private final BooleanSupplier condition;
+    @Nullable
+    private final String message;
+    private final boolean isError;
+
+    Posession(@NotNull BooleanSupplier condition) {
+      this.condition = condition;
+      this.message = null;
+      isError = false;
+    }
+
+    Posession(@NotNull BooleanSupplier condition, @Nullable String message, boolean isError) {
+      this.condition = condition;
+      this.message = message;
+      this.isError = isError;
+    }
+
+  }
+
+  private enum MotionProfile {
+    GO_STRAIGHT(new SimpleProfileAuto("go_straight"), "Crossing the line"),
+    LEFT_HOOK(new SingleCubeSwitchAuto("left_hook"), "Going for the left switch"),
+    LEFT_SCALE(new ProfileScaleAuto("left_scale"), "Going for the left scale"),
+    LEFT_SCALE_NO_SWITCH(new ProfileDoubleScaleAuto("left_scale", "left_scale_back_to_switch", "left_switch_back_to_scale"), "Going for the left scale"),
+    LEFT_DOUBLE_SCALE_THEN_SWITCH(new ProfileDoubleScaleAuto("left_scale_straight", "left_scale_straight_back_to_switch", "left_switch_straight_back_to_scale"), "Going for the left scale"),
+    LEFT_SCALE_STRAIGHT(new ProfileDoubleScaleAuto("left_scale_straight", "left_scale_straight_back_to_switch", "left_switch_straight_back_to_scale"), "Going for left scale"),
+    MIDDLE_TO_LEFT_SWITCH(new SingleCubeSwitchAuto("middle_to_left_switch"), "Going for the left switch"),
+    MIDDLE_TO_RIGHT_SWITCH(new SingleCubeSwitchAuto("middle_to_right_switch"), "Going for the right switch"),
+    SWITCH_DOUBLE_CUBE_LEFT(new SwitchDoubleCube("left"), "Going for the left switch"),
+    SWITCH_DOUBLE_CUBE_RIGHT(new SwitchDoubleCube("right"), "Going for the right switch"),
+    RIGHT_HOOK(new SingleCubeSwitchAuto("right_hook"), "Going for the right switch"),
+    RIGHT_SCALE(new ProfileScaleAuto("right_scale"), "Going for the right scale"),
+    DRIVE_TIMED(new DriveTimed(ControlMode.PercentOutput, Tuning.stupidDriveTime, Tuning.stupidDrivePercent), "Driving straight for time");
+
+    private final Command autoCommand;
+    private final String defaultMessage;
+
+    MotionProfile(Command autoCommand, String defaultMessage) {
+      this.autoCommand = autoCommand;
+      this.defaultMessage = defaultMessage;
+    }
+
+  }
+
+  private static class DecisionNode {
+
+    @NotNull
+    final BooleanSupplier condition;
+    @Nullable
+    private Message message = null;
+    @Nullable
+    final MotionProfile profile;
+    @Nullable
+    final DecisionNode[] children;
+
+    public DecisionNode(Posession posession, @NotNull MotionProfile profile) {
```

Comment:
`DecisionNodes` taking a `Possession` is unnecessary, just let them take a generic `BooleanSupplier` and provide a couple of static final ones corresponding to switch/scale condition.

Reply:
I had it written that way, but this is cleaner for writing. If we decide to change that, it's trivial to make a generic version.